cmd/grpcurl/grpcurl.go

```diff
@@ -652,7 +663,11 @@ func main() {
 			fmt.Printf("Sent %d request%s and received %d response%s\n", reqCount, reqSuffix, h.NumResponses, respSuffix)
 		}
 		if h.Status.Code() != codes.OK {
-			grpcurl.PrintStatus(os.Stderr, h.Status, formatter)
+			if *formatError {
+				printFormattedStatus(os.Stderr, h.Status, formatter)
+			} else {
+				grpcurl.PrintStatus(os.Stderr, h.Status, formatter)
+			}
 			exit(1)
```

Comment:
Does it make sense for the `exit(1)` here to use the response status code instead: `exit(int(h.Status.Code()))`?

https://github.com/fullstorydev/grpcurl/blob/8243c87e864cbe2ad959d1e0ecdd26576258657b/cmd/grpcurl/grpcurl.go#L671

Reply:
I like that idea!

But I think we should use `64+code` to avoid confusion as to whether between an exit code of 1 or 2 means `Canceled` or `Unknown` vs. some other error. This program already uses exit code 1 to mean unexpected error (like failures that prevented an RPC from even being sent, such as specifying bad filenames, being unable to dial server, etc). And we also use exit code 2 to mean incorrect usage (bad combination of command-line args).

So to avoid confusion with those scenarios, the range 64+ should be good to mean that RPC was actually performed but ultimately failed.

I suggest 64 as the constant (as opposed to say 4, 8, 16, or 32) since that is the range recommended by a resource I found on the web for user-defined exit codes: http://www.tldp.org/LDP/abs/html/exitcodes.html
Seems like reasonable advice...